ztp/ran-site-plan-crd/site-config-crd.yaml

```diff
@@ -0,0 +1,298 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  # name must match the spec fields below, and be in the form: <plural>.<group>
+  name: siteconfig.ran.openshift.io
+spec:
+  # group name to use for REST API: /apis/<group>/<version>
+  group: ran.openshift.io
+  scope: Namespaced
+  names:
+    # plural name to be used in the URL: /apis/<group>/<version>/<plural>
+    plural: siteconfigs
+    # singular name to be used as an alias on the CLI and for display
+    singular: siteconfig
+    # kind is normally the CamelCased singular type. Your resource manifests use this.
+    kind: SiteConfig
+    # shortNames allow shorter string to match your resource on the CLI
+    shortNames:
+    - SiteConfig
+  # list of versions supported by this CustomResourceDefinition
+  versions:
+    - name: v1
+      # Each version can be enabled/disabled by Served flag.
+      served: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            spec:
+              type: object
+              properties:
+                baseDomain:
+                  description: BaseDomain is the base domain to which the cluster should belong.
+                  type: string
+                pullSecret:
+                  description: PullSecret to use when pulling images.
+                  type: string
+                releaseImageUrl:
+                  description: |
+                    releaseImageUrl is the url to the image containing metadata for all
+                    components that run in the cluster, and is the primary and best way
+                    to specify what specific version of OpenShift you wish to install.
+                  type: string
+                sshPublicKey:
+                  description: sshPublicKey will be added to all cluster hosts for use in debugging.
+                  type: string
+                sshPrivateKey:
```

Comment:
Sensitive items like this (and the BMC password, and pull secret too) should be references to secrets, and not stored plaintext.

Reply:
the idea is to create the secret , so the user provide data in base64 encoded txt then we create the secret and other stuff

Reply:
But wouldn't this mean the user is putting their passwords in clear text in git?

Reply:
well, it is not clear text it is base64 encoded :D but this is even how k8s store the secrets

Reply:
The difference though is that k8s restricts access to the secrets by RBAC roles; encouraging storing of secrets in git is bad security practice.

Reply:
I agree, its just user will have to provide secret names instead of the bmc info.

Reply:
- For the SSH key specifically, can we just generate a private key and inject the associated public key without having to specify any of it in git? (And we would also inject additional SSH public key(s) specified by the user for their own reasons)
- For the BMC password, we should definitely use a secret instead of putting that info in git.  With that password, an attacker would own the machine entirely.
- The pull-secret contents is also protected and should maybe be protected in a secret as well?

Maybe @yuvalk  and @dbaker-rh can chime in with their thoughts on the security aspects of this CRD, given that we expect end-users to check their associated CRs into git?

Reply:
+1 on having all keys and passwords as secret references.
BTW, in this case SSH public key appearing as a string is inherited (I believe) from AI CRD. Probably need to have this raised there? Or is it OK to list public keys in clear text?

Reply:
Yes, public keys are safe to expose in clear text.

@serngawy : Can you help us understand why we need a private key in this information at all?